### PR TITLE
Going with Apache Kafka 3.1

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

### CODE REVIEW

**Comment by matzew** [comment]
```
    knative_eventing_test.go:34: Failed to verify that health metrics work correctly for Eventing failed to access the Prometheus API endpoint and get the metric value expected: timed out waiting for the condition
```

/retest

File: hack/lib/strimzi.bash [hunk 1]
```diff
@@ -48,9 +48,7 @@ function install_strimzi_cluster {
           offsets.topic.replication.factor: 3
           transaction.state.log.replication.factor: 3
           transaction.state.log.min.isr: 2
-          inter.broker.protocol.version: "3.0"
-          log.message.format.version: "3.0"
-          auto.create.topics.enable: "false"
```
Comment (pierDipi) [inline hack/lib/strimzi.bash:53]: is this setting gone? it's nice to have that one to make sure we're actually creating topics.

Comment (matzew) [reply hack/lib/strimzi.bash:53]: >           auto.create.topics.enable: "false"

jikes! I removed that by accident.

Added it back

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**